system-tests/lib/cre/types.go

```diff
@@ -512,51 +512,71 @@ type DonWithMetadata struct {
 }
 
 type DonMetadata struct {
-	NodesMetadata   []*NodeMetadata `toml:"nodes_metadata" json:"nodes_metadata"`
-	Flags           []string        `toml:"flags" json:"flags"`
-	ID              uint64          `toml:"id" json:"id"`
-	Name            string          `toml:"name" json:"name"`
-	SupportedChains []uint64        `toml:"supported_chains" json:"supported_chains"` // chain IDs that the DON supports, empty means all chains
+	NodesMetadata []*NodeMetadata `toml:"nodes_metadata" json:"nodes_metadata"`
+	Flags         []string        `toml:"flags" json:"flags"`
+	ID            uint64          `toml:"id" json:"id"`
+	Name          string          `toml:"name" json:"name"`
+	//	EVMChainIDs   []uint64        `toml:"supported_chains" json:"supported_chains"` // chain IDs that the DON supports, empty means all chains
 
 	ns CapabilitiesAwareNodeSet // computed field, not serialized
 }
 
-func NewDonMetadata(c *CapabilitiesAwareNodeSet, id uint64) *DonMetadata {
+func NewDonMetadata(c *CapabilitiesAwareNodeSet, id uint64) (*DonMetadata, error) {
+
+	// generate keys for each node
+	evmChains := make([]int, len(c.EVMChains()))
```

Comment:
how will that work with a situation, when we use TOML config, where keys/secrets are already present? In that situation we don't want to generate any keys. `NewDonMetadata()` sounds like an universal constructor that should handle both cases.